### PR TITLE
Add binding for LPS22HB

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Add class, sample and nuspec for publishing.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: devices/Lps22Hb/FifoMode.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        /// <summary>
+        /// Bypass mode.
+        /// </summary>
+        Bypass = 0,
+
+        /// <summary>
+        /// FIFO mode.
+        /// </summary>
+        FIFO = 1,
```
Comment (Ellerbach) [inline devices/Lps22Hb/FifoMode.cs:19]: Maybe `Fifo`?

File: devices/Lps22Hb/Lps22Hb.cs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+        /// <summary>
+        /// Configuration: ODR = 25 Hz(continuous mode), LPF active with ODR/9, BDU activ.
+        /// </summary>
+        private const int ConfigFifoBypass = 0x3A;
+
+        /// <summary>
+        /// Device ID when reading the WHO_AM_I register.
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x5C;
```
Comment (Ellerbach) [inline devices/Lps22Hb/Lps22Hb.cs:29]: the description is not correct and doesn't match with the one line 45.

File: devices/Lps22Hb/Lps22Hb.cs [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        /// <summary>
+        /// This holds the magnetic field reading from the sensor.
+        /// Elements index is 0 for X, 1 for Y and 2 for Z.
+        /// </summary>
+        private float[] _magneticFieldReadings = new float[3];
+
+        /// <summary>
+        /// Device I2C Address.
+        /// </summary>
+        public const byte I2cAddress = 0x1E;
```
Comment (Ellerbach) [inline devices/Lps22Hb/Lps22Hb.cs:45]: is this WhoAmI or the DefaultI2cAddress ? :-)

File: devices/Lps22Hb/Lps22Hb.cs [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        /// <exception cref="Exception">If the device is not found.</exception>
+        public Lps22Hb(
+            I2cDevice i2cDevice,
+            FifoMode mode)
+        {
+            _i2c = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            // check if the device is present
+            var id = Read(Register.WhoAmI);
+            if (id != DefaultI2cAddress)
```
Comment (Ellerbach) [inline devices/Lps22Hb/Lps22Hb.cs:62]: wrong rename ;-)

File: devices/Lps22Hb/Lps22Hb.cs [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+                case FifoMode.FIFO:
+                case FifoMode.Stream:
+                case FifoMode.StreamToFifo:
+                case FifoMode.BypassToStream:
+                case FifoMode.BypassToFifo:
+                case FifoMode.DynamicStream:
+                default:
+                    // not implemented
+                    throw new NotImplementedException();
```
Comment (Ellerbach) [inline devices/Lps22Hb/Lps22Hb.cs:174]: why are all those modes not supported?

File: devices/Lps22Hb/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    var tempValue = lps22HdDevice.Temperature;
+    var pressure = lps22HdDevice.Pressure;
+
+    Debug.WriteLine($"Temperature: {tempValue.DegreesCelsius:F1}\u00B0C");
+    Debug.WriteLine($"Pressure: {pressure.Hectopascals:F1}hPa");
+
+    Thread.Sleep(1000);
+}
+
+I2cDevice CreateI2cDevice()
```
Comment (Ellerbach) [inline devices/Lps22Hb/README.md:47]: one of the 2 functions are not needed :-)

**Review by Ellerbach — COMMENTED** [review]
Couple of things to adjust as one of the rename is not the correct one :-)

File: devices/Lps22Hb/Lps22Hb.nuspec [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    <copyright>Copyright (c) .NET Foundation and Contributors</copyright>
+    <description>This package includes the .NET IoT Core binding Iot.Device.Lps22Hb for .NET nanoFramework C# projects.</description>
+    <summary>Iot.Device.Lps22Hb assembly for .NET nanoFramework C# projects</summary>
+    <tags>nanoFramework C# csharp netmf netnf Iot.Device.Lps22Hb</tags>
+    <dependencies>
+      <dependency id="nanoFramework.CoreLibrary" version="1.15.5" />
+      <dependency id="nanoFramework.System.Device.Model" version="1.2.418" />
+      <dependency id="nanoFramework.System.Device.I2c" version="1.1.16" />
+      <dependency id="nanoFramework.System.Buffers.Binary.BinaryPrimitives" version="1.2.418" />
+      <dependency id="UnitsNet.nanoFramework.Temperature" version="5.48.0" />
```
Comment (Ellerbach) [inline devices/Lps22Hb/Lps22Hb.nuspec:26]: I think the build is red because you may be missing math here

**Review by Ellerbach — APPROVED** [review]
Looks good now, still something missing in the nuget I think

File: devices/Lps22Hb/Lps22Hb.nuspec (hunk 7 above)
Comment (josesimoes) [reply devices/Lps22Hb/Lps22Hb.nuspec:26]: it's a dependency of Pressure, therefore it doesn't need to be there

**Pull request merged**